src/ripple/basics/base_uint.h

```diff
@@ -363,99 +373,62 @@ class base_uint
     }
 
     /** Parse a hex string into a base_uint
-        The string must contain exactly bytes * 2 hex characters and must not
-        have any leading or trailing whitespace.
-    */
-    bool
-    SetHexExact(const char* psz)
-    {
-        unsigned char* pOut = begin();
 
-        for (int i = 0; i < sizeof(data_); ++i)
-        {
-            auto hi = charUnHex(*psz++);
-            if (hi == -1)
-                return false;
-
-            auto lo = charUnHex(*psz++);
-            if (lo == -1)
-                return false;
-
-            *pOut++ = (hi << 4) | lo;
-        }
+        The input must be precisely `2 * bytes` hexadecimal characters
+        long, with one exception: the value '0'.
 
-        // We've consumed exactly as many bytes as we needed at this point
-        // so we should be at the end of the string.
-        return (*psz == 0);
-    }
-
-    /** Parse a hex string into a base_uint
-        The input can be:
-            - shorter than the full hex representation by not including leading
-              zeroes.
-            - longer than the full hex representation in which case leading
-              bytes are discarded.
-
-        When finished parsing, the string must be fully consumed with only a
-        null terminator remaining.
-
-        When bStrict is false, the parsing is done in non-strict mode, and, if
-        present, leading whitespace and the 0x prefix will be skipped.
-    */
-    bool
-    SetHex(const char* psz, bool bStrict = false)
+        @param sv A null-terminated string of hexadecimal characters
+        @return true if the input was parsed properly; false otherwise.
+     */
+    [[nodiscard]] bool
+    parseHex(std::string_view sv)
```

Comment:
`parseHex` is a replacement for `SetHexExact`, but not for the more lenient `SetHex`. This breaks backward compatibility in many RPC functions. It's OK with me if we want to do that, but we would definitely document the change and note the break with backward compatibility. 

I'll tag all the places I found that would break backward compatibility in case it helps with updating docs.